### DISCUSSION TITLE
Paging feeds getting stuck in a loop

### DISCUSSION THREAD

**Discussion by bburkle-cmt** [transitland/transitland#402] [Q&A] [answered]
I'm using the Transitland v2 Rest API and querying`/api/v2/rest/feeds` with the following request:
`https://transit.land/api/v2/rest/feeds?apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`

I am then taking the `next` field of the `meta` response to go to the next page as described [here](https://www.transit.land/documentation/rest-api/#pagination) to access the next page. However, after 6 pages the `next` field keeps repeating the page that I am currently on, causing my scripts to get stuck in a loop continuously calling the same page. The order of urls that I am getting taken to is:

- `https://transit.land/api/v2/rest/feeds?after=25&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`
- `https://transit.land/api/v2/rest/feeds?after=28&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`
- `https://transit.land/api/v2/rest/feeds?after=30&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`
- `https://transit.land/api/v2/rest/feeds?after=32&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`
- `https://transit.land/api/v2/rest/feeds?after=33&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`
- `https://transit.land/api/v2/rest/feeds?after=34&apikey=my-api-key&license_commercial_use_allowed=exclude_no&license_create_derived_product=exclude_no&license_redistribution_allowed=exclude_no&spec=GTFS`

Once on `after=34` the `meta` response then points me back at the same page. I've gone to the url in my browser to confirm that this is a problem with the response and not the way that I am ingesting it. I also tried  to modify the url to try to take me to a later page, but it still points me back to the page with `after=34` in the `meta` response.

**Comment by irees** [comment, marked as answer]
hi @bburkle-cmt I've identified the problem (a missing set of parenthesis in a complex 'where' statement on the license filtering clauses) and will have a fix deployed soon.

**Reply by bburkle-cmt** [reply]
Great! Thanks a lot for finding the cause so quickly!

**Reply by irees** [reply]
The fix is now deployed to production

**Reply by bburkle-cmt** [reply]
Thank you again for taking care of this!
